src/ast/rewriter/seq_rewriter.cpp

```diff
@@ -4527,16 +4527,16 @@ br_status seq_rewriter::mk_str_in_regexp(expr* a, expr* b, expr_ref& result) {
         return BR_REWRITE_FULL;
     }
 
-    if (get_re_head_tail(b, hd, tl)) {
-        SASSERT(re().min_length(hd) == re().max_length(hd));
-        expr_ref len_hd(m_autil.mk_int(re().min_length(hd)), m()); 
-        expr_ref len_a(str().mk_length(a), m());
-        expr_ref len_tl(m_autil.mk_sub(len_a, len_hd), m());
-        result = m().mk_and(m_autil.mk_ge(len_a, len_hd),
-                            re().mk_in_re(str().mk_substr(a, zero(), len_hd), hd),
-                            re().mk_in_re(str().mk_substr(a, len_hd, len_tl), tl));
-        return BR_REWRITE_FULL;
-    }
+    // if (get_re_head_tail(b, hd, tl)) {
+    //     SASSERT(re().min_length(hd) == re().max_length(hd));
+    //     expr_ref len_hd(m_autil.mk_int(re().min_length(hd)), m()); 
+    //     expr_ref len_a(str().mk_length(a), m());
+    //     expr_ref len_tl(m_autil.mk_sub(len_a, len_hd), m());
+    //     result = m().mk_and(m_autil.mk_ge(len_a, len_hd),
+    //                         re().mk_in_re(str().mk_substr(a, zero(), len_hd), hd),
+    //                         re().mk_in_re(str().mk_substr(a, len_hd, len_tl), tl));
+    //     return BR_REWRITE_FULL;
+    // }
```

Comment:
why is this commented out?

Reply:
This seems to be the code that performs replacing of certain regexes to `substr` predicate. This is not beneficial for us because we aim at the regular constraints.

Reply:
Maybe just change the first line to `if (false && get_re_head_tail(b, hd, tl)) { \\FIXME-NOODLER: replacing regexes to substr is not beneficial for noodler` so we change as little lines of z3 code. It would help when we will want to update the z3 version for noodler.
Or maybe, is it possible to check here if str solver is set to noodler? So the condition would be ``if (!str_solver_is_noodler && get_re_head_tail(b, hd, tl))`.

Reply:
We should probably add something like `FIXME: noodler substring temp fix` so that we can find this place later.

Reply:
you are right

Reply:
I am not aware of the second option. I guess it could work, but it would require some effort